packages/nodes-base/nodes/Wise/Wise.node.ts

```diff
@@ -143,10 +144,11 @@ export class Wise implements INodeType {
 
 				const recipients = await wiseApiRequest.call(this, 'GET', 'v1/accounts', {}, qs);
 
-				return recipients.map(({ id, accountHolderName }: Recipient) => ({
-					name: accountHolderName,
-					value: id,
-				}));
+				return recipients.filter(({active} : Recipient) => active)
+					.map(({ id, accountHolderName, currency, country, type }: Recipient) => ({
+						name: `[${currency}] ${accountHolderName} - (${country !== null ? country + ' - ' : '' }${type})`,
+						value: id,
+					}));
```

Comment:
`reduce` can do this in a single pass.

Reply:
`as Recipient[]` at L145 so the item type is inferrable in the loop.

Reply:
`country !== null` will always be true according to the typing. Either the typing or the check needs adjusting.

https://github.com/n8n-io/n8n/blob/189a6e61e64921f5e193c7f0e43a1b8fa57c593e/packages/nodes-base/nodes/Wise/GenericFunctions.ts#L209-L216

Reply:
Thanks, always have trouble conceptualising reduce. Let me know if the updated change is what you expected.

Reply:
Exactly. 

Minor optional suggestions:
- The type of the previous value should be provided [as a generic](https://github.com/microsoft/TypeScript/blob/afee8bf9682dfb3ed927fdc7756d96a19851b811/lib/lib.es5.d.ts#L1386 ) to `reduce`
- The type of the current value can be omitted as it is inferrable from the array
- In `Recipient`, `currency` should be `string | null`  for the check to make sense

```ts
return recipients.reduce<INodePropertyOptions[]>((activeRecipients, {
	active,
	id,
	accountHolderName,
	currency,
	country,
	type,
}) => {
	if (active) {
		const recipient = {
			name: `[${currency}] ${accountHolderName} - (${country !== null ? country + ' - ' : '' }${type})`,
			value: id,
		};
		activeRecipients.push(recipient);
	}
	return activeRecipients;
}, []);
```

Reply:
Exactly.

Minor optional suggestions:
- The type for the accumulator is provided [as a generic](https://github.com/microsoft/TypeScript/blob/afee8bf9682dfb3ed927fdc7756d96a19851b811/lib/lib.es5.d.ts#L1386) to `reduce`
- The type for the current value can be omitted as it is inferrable from the array
- In `Recipient`, `country` should be `string | null` for the check to make sense.

```ts
return recipients.reduce<INodePropertyOptions[]>((activeRecipients, {
	active,
	id,
	accountHolderName,
	currency,
	country,
	type,
}) => {
	if (active) {
		const recipient = {
			name: `[${currency}] ${accountHolderName} - (${country !== null ? country + ' - ' : '' }${type})`,
			value: id,
		};
		activeRecipients.push(recipient);
	}
	return activeRecipients;
}, []);
```

Reply:
Amazing, I have some reading to do on generics. Have tested and updated the commit with your suggestions.